src/Blocks/BlockCauldron.h

```diff
@@ -98,6 +99,65 @@ class cBlockCauldronHandler :
 						a_Player.ReplaceOneEquippedItemTossRest(cItem(E_ITEM_GLASS_BOTTLE));
 					}
 				}
+				break;
+			}
+			// Resets any color to default:
+			case E_ITEM_LEATHER_BOOTS:
+			case E_ITEM_LEATHER_CAP:
+			case E_ITEM_LEATHER_PANTS:
+			case E_ITEM_LEATHER_TUNIC:
+			{
+				if ((Meta > 0) && ((EquippedItem.m_ItemColor.GetRed() != 255) || (EquippedItem.m_ItemColor.GetBlue() != 255) || (EquippedItem.m_ItemColor.GetGreen() != 255)))
+				{
+					a_ChunkInterface.SetBlockMeta(a_BlockPos, --Meta);
+					auto NewItem = cItem(EquippedItem);
+					NewItem.m_ItemColor.Clear();
+					a_Player.ReplaceOneEquippedItemTossRest(NewItem);
+				}
+				break;
+			}
+			// Resets shulker box color:
+			case E_BLOCK_BLACK_SHULKER_BOX:
+			case E_BLOCK_BLUE_SHULKER_BOX:
+			case E_BLOCK_BROWN_SHULKER_BOX:
+			case E_BLOCK_CYAN_SHULKER_BOX:
+			case E_BLOCK_GRAY_SHULKER_BOX:
+			case E_BLOCK_GREEN_SHULKER_BOX:
+			case E_BLOCK_LIGHT_BLUE_SHULKER_BOX:
+			case E_BLOCK_LIGHT_GRAY_SHULKER_BOX:
+			case E_BLOCK_LIME_SHULKER_BOX:
+			case E_BLOCK_MAGENTA_SHULKER_BOX:
+			case E_BLOCK_ORANGE_SHULKER_BOX:
+			case E_BLOCK_PINK_SHULKER_BOX:
+			case E_BLOCK_RED_SHULKER_BOX:
+			case E_BLOCK_YELLOW_SHULKER_BOX:
+			{
+				// TODO: When there is an actual default shulker box add the appropriate changes here! - 19.09.2020 - 12xx12
+				if (Meta > 0)
+				{
+					// This is a workaround for version < 1.13. They client thinks a player placed a shulker and display that to the player
+					// The shulker cleaning was added in 1.13.
+					if (a_Player.GetPosition().y != a_BlockPos.addedY(1).y)  // Player is not standing on top
+					{
+						auto NewPos = a_BlockPos;
+						switch (a_BlockFace)
```

Comment:
Could AddFaceDirection work here?

Reply:
I knew there was a function like that. I tried to find this function...
Whats the class for that function

Reply:
It should be a free function, in Defines.h

Reply:
ah i looked for that in the vector3i class